### ISSUE TITLE
There is an error in the compilerconfig.json file

### ISSUE THREAD

**Issue by JBowness** [madskristensen/WebCompiler#128]
I have two .scss files in an existing project.  When I save the first one, it automatically creates the compilerconfig.json file and adds outputFile and inputFile settings for the scss/css.  The sass compiles to CSS, and all is well.

Then I edit a 2nd .scss file.  It adds the appropriate outputFile and inputFile settings to the compilerconfig.json file, and all appears well, until I compile.  I get an exception message that reads "There is an error in the compilerconfig.json file. This could be due to a change in the format after this extension was updated."

From what I can see, this is just a generic exception during build.  I'm confident that the auto-generated json syntax is correct.  Can you recommend a way to get to the actual exception that was logged?   Thanks kindly!

**Comment by madskristensen** [comment]
Can you check the output window?

**Comment by JBowness** [comment]
Sorry, that is more helpful.   Here's the output:


10/13/2015 12:05:53 PM: System.ArgumentNullException: Value cannot be null.
Parameter name: uriString
   at System.Uri..ctor(String uriString, UriKind uriKind)
   at WebCompiler.FileHelpers.MakeRelative(String baseFile, String file)
   at WebCompiler.CssRelativePath.Adjust(String content, Config config)
   at WebCompiler.ConfigFileProcessor.ProcessConfig(String baseFolder, Config config)
   at WebCompiler.ConfigFileProcessor.Process(String configFile, IEnumerable`1 configs)
   at WebCompilerVsix.CompilerService.<>c__DisplayClass6_0.<Process>b__0(Object o) in C:\projects\webcompiler\src\WebCompilerVsix\CompilerService.cs:line 61

**Comment by madskristensen** [comment]
Can you post a link to the 2nd .scss file so I can debug the issue?

**Comment by madskristensen** [comment]
Also, make sure you are using the latest version of the extension. Grab it here http://vsixgallery.com/extension/148ffa77-d70a-407f-892b-9ee542346862/

**Comment by JBowness** [comment]
I'm wondering if it has something to do with the mixins living in a different path than the scss?

I don't think IIS will serve the scss, but here's the relevant compilerconfig and the 2nd scss:

[
  {
    "outputFile": "App_Templates/Skin_1/GeneratedCSS/app.css",
    "inputFile": "App_Templates/Skin_1/scss/app.scss"
  },
  {
    "outputFile": "Dogs/assets/css/app.css",
    "inputFile": "Dogs/assets/sass/app.scss"
  }
]


And the contents of Dogs/assets/sass/app.scss:


// Core variables and mixins
@import "bootstrap/variables";
@import "bootstrap/mixins";

// Reset and dependencies
@import "bootstrap/normalize";
@import "bootstrap/print";
@import "bootstrap/glyphicons";

// Core CSS
@import "bootstrap/scaffolding";
@import "bootstrap/type";
@import "bootstrap/code";
@import "bootstrap/grid";

// Utility classes
@import "bootstrap/utilities";
@import "bootstrap/responsive-utilities";

//Custom
@import "base/fonts";
@import "base/base";

//Pages
@import 'app/header';
@import 'app/top';
@import 'app/8-content';
@import 'app/paws';
@import 'app/foreground-float';
@import 'app/circle-images';
@import 'app/parallax';

/* Medium devices (desktops, 992px and up) */
@media (min-width: $screen-md-min) {}

@media (min-width: $screen-sm-min) {}

**Comment by madskristensen** [comment]
There has been issues with fixing relative urls in the past, but they should be fixed in the latest nightly. Can you please download that build and verify that the problem is fixed? http://vsixgallery.com/extension/148ffa77-d70a-407f-892b-9ee542346862/

**Comment by JBowness** [comment]
Thanks.  Tried that version, and still getting the behavior:

10/14/2015 11:49:03 AM: System.ArgumentNullException: Value cannot be null.
Parameter name: uriString
   at System.Uri..ctor(String uriString, UriKind uriKind)
   at WebCompiler.FileHelpers.MakeRelative(String baseFile, String file) in C:\projects\webcompiler\src\WebCompiler\Helpers\FileHelpers.cs:line 17
   at WebCompiler.CssRelativePath.Adjust(String content, Config config) in C:\projects\webcompiler\src\WebCompiler\Compile\CssRelativePathAdjuster.cs:line 13
   at WebCompiler.ConfigFileProcessor.ProcessConfig(String baseFolder, Config config) in C:\projects\webcompiler\src\WebCompiler\Config\ConfigFileProcessor.cs:line 184
   at WebCompiler.ConfigFileProcessor.Process(String configFile, IEnumerable`1 configs, Boolean force) in C:\projects\webcompiler\src\WebCompiler\Config\ConfigFileProcessor.cs:line 44
   at WebCompilerVsix.CompilerService.<>c__DisplayClass6_0.<Process>b__0(Object o) in C:\projects\webcompiler\src\WebCompilerVsix\CompilerService.cs:line 60


Is there anything I can do to help with diagnosing?

**Comment by madskristensen** [comment]
You already did. This stack trace has more precise information than the last due to the latest CI build, so now I know what to do

**Comment by JBowness** [comment]
Fantastic!  Looking forward to the nightly build.  Thanks most kindly!

**Comment by madskristensen** [comment]
It should be ready now on the same URL

**Comment by JBowness** [comment]
Thanks Mads.  Getting further, but now seeing the following, even when minify is not enabled:

10/14/2015 1:29:34 PM: System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
   at Microsoft.Ajax.Utilities.CssScanner.GetString()
   at Microsoft.Ajax.Utilities.CssScanner.ScanUrl()
   at Microsoft.Ajax.Utilities.CssScanner.NextToken()
   at Microsoft.Ajax.Utilities.CssParser.NextToken()
   at Microsoft.Ajax.Utilities.CssParser.SkipSpace()
   at Microsoft.Ajax.Utilities.CssParser.ParseDeclaration()
   at Microsoft.Ajax.Utilities.CssParser.ParseDeclarationList(Boolean allowMargins)
   at Microsoft.Ajax.Utilities.CssParser.ParseDeclarationBlock(Boolean allowMargins)
   at Microsoft.Ajax.Utilities.CssParser.ParseRule()
   at Microsoft.Ajax.Utilities.CssParser.ParseStylesheet()
   at Microsoft.Ajax.Utilities.CssParser.Parse(String source)
   at Microsoft.Ajax.Utilities.Minifier.MinifyStyleSheet(String source, CssSettings settings, CodeSettings scriptSettings)
   at WebCompiler.FileMinifier.MinifyCss(Config config, String file) in C:\projects\webcompiler\src\WebCompiler\Minify\FileMinifier.cs:line 76
   at WebCompiler.FileMinifier.MinifyFile(Config config) in C:\projects\webcompiler\src\WebCompiler\Minify\FileMinifier.cs:line 25
   at WebCompiler.ConfigFileProcessor.ProcessConfig(String baseFolder, Config config) in C:\projects\webcompiler\src\WebCompiler\Config\ConfigFileProcessor.cs:line 156
   at WebCompiler.ConfigFileProcessor.Process(String configFile, IEnumerable`1 configs, Boolean force) in C:\projects\webcompiler\src\WebCompiler\Config\ConfigFileProcessor.cs:line 44
   at WebCompilerVsix.CompilerService.<>c__DisplayClass6_0.<Process>b__0(Object o) in C:\projects\webcompiler\src\WebCompilerVsix\CompilerService.cs:line 60

**Comment by madskristensen** [comment]
Is there a way you can send me your project or even better if you could make a zip file of _smallest repro possible_? I need to debug this with the actual failing sources

**Comment by JBowness** [comment]
Here's a slimmed down version which is choking for me. 
https://www.dropbox.com/s/txugf51kcleycf2/Dogs.zip?dl=0

**Comment by madskristensen** [comment]
This happens in AjaxMin when it runs minification of the correctly compiled CSS ouput. Very strange...

**Comment by JBowness** [comment]
Yes, it appears to generate the css fine, but does afterwards... Even when
minification is disabled

**Comment by madskristensen** [comment]
There is a control character in the output right after `.jpg`:

![screenshot_1](https://cloud.githubusercontent.com/assets/1258877/10499507/8fb8263a-7285-11e5-8d0a-ced131fcfebe.png)

I can remove those before I run AjaxMin, but perhaps for now you can remove them from the .scss source?

**Comment by JBowness** [comment]
I tried a search and replace on the project for this control character, and
it eliminated some... but upon saving the resulting files, it crashed
again, and this time took the IDE with it.

I get there's only so much you can do when dealing with unexpected
characters, but I'm at a loss how these characters were introduced between
my designer on his mac to me on my Win 10 PC.

I'll keep trying to clean the files to get a proper compile.

**Comment by JBowness** [comment]
Does AjaxMin run when minify is disabled?   I'm getting the generated css file written, but then it continues to do stuff in the background, resulting in laggy IDE and eventually the exception.  This is after I've cleaned my project of the control character.

Would you like a cleaned repro project?

10/14/2015 3:39:05 PM: System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
   at System.Text.StringBuilder.ToString()
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(IFormatProvider provider, String format, Object[] args)
   at Microsoft.Ajax.Utilities.AjaxMinExtensions.FormatInvariant(String format, Object[] args)
   at Microsoft.Ajax.Utilities.CssScanner.ReportError(Int32 severity, CssErrorCode error, Object[] args)
   at Microsoft.Ajax.Utilities.CssScanner.GetString()
   at Microsoft.Ajax.Utilities.CssScanner.ScanUrl()
   at Microsoft.Ajax.Utilities.CssScanner.NextToken()
   at Microsoft.Ajax.Utilities.CssParser.NextToken()
   at Microsoft.Ajax.Utilities.CssParser.SkipSpace()
   at Microsoft.Ajax.Utilities.CssParser.ParseDeclaration()
   at Microsoft.Ajax.Utilities.CssParser.ParseDeclarationList(Boolean allowMargins)
   at Microsoft.Ajax.Utilities.CssParser.ParseDeclarationBlock(Boolean allowMargins)
   at Microsoft.Ajax.Utilities.CssParser.ParseRule()
   at Microsoft.Ajax.Utilities.CssParser.ParseStylesheet()
   at Microsoft.Ajax.Utilities.CssParser.Parse(String source)
   at Microsoft.Ajax.Utilities.Minifier.MinifyStyleSheet(String source, CssSettings settings, CodeSettings scriptSettings)
   at WebCompiler.FileMinifier.MinifyCss(Config config, String file) in C:\projects\webcompiler\src\WebCompiler\Minify\FileMinifier.cs:line 76
   at WebCompiler.FileMinifier.MinifyFile(Config config) in C:\projects\webcompiler\src\WebCompiler\Minify\FileMinifier.cs:line 25
   at WebCompiler.ConfigFileProcessor.ProcessConfig(String baseFolder, Config config) in C:\projects\webcompiler\src\WebCompiler\Config\ConfigFileProcessor.cs:line 156
   at WebCompiler.ConfigFileProcessor.SourceFileChanged(String configFile, String sourceFile) in C:\projects\webcompiler\src\WebCompiler\Config\ConfigFileProcessor.cs:line 76
   at WebCompilerVsix.CompilerService.<>c__DisplayClass7_0.<SourceFileChanged>b__0(Object o) in C:\projects\webcompiler\src\WebCompilerVsix\CompilerService.cs:line 98

**Comment by JBowness** [comment]
Actually, is minification enabled by default?  Even when minify : {enabled : false})

I appear to be able to get by when I disable minification.  I'd incorrectly assumed that not specifying minify would skip minification.

**Comment by madskristensen** [comment]
Yes, on by default

**Comment by madskristensen** [comment]
The latest nightly now have this fixed. Minification is enabled by default. You can turn it off in the `compilerconfig.json.defaults` for your entire project

**Comment by JBowness** [comment]
Perfect!  Working like a dream now!

I'm guessing there's another control character in there somewhere that I didn't find, but this update has me sailing.  Thanks so much.

**Comment by zgrkpnr** [comment]
@madskristensen I would like to add something. In my case, the same error was caused by TFS. You need to check-out less and css files before compiling it. Becaue, without checking out TFS doesn't allow any editing on files.
